### DISCUSSION TITLE
Dynamic Link

### DISCUSSION THREAD

**Discussion by pme123** [planet42/Laika#318] [Q&A] [answered]
Is there a way to create a Web-Link dynamically, for example
`[JIRA](https://myJira/projects/myProject/versions/${jira.release.id})`
But `${jira.release.id}` is not resolved - if I add just `${jira.release.id}` it is.

Is there a way to achieve this?

**Comment by jenshalm** [comment, marked as answer]
The variable substitution you tried is mostly just supported on the top level of markup sources, not as part of syntax that is specific to a particular markup format such as Markdown. I agree that it would be neat if the above would just work, but it's non-trivial to support, which means it's out of scope for 0.19.

In the meantime you could try link directives instead. They require a bit of setup, but at least they are also more convenient when you use the same approach multiple times, as you only have to define the base URL in one place. Details about link directives can be found here: https://planet42.github.io/Laika/0.18/05-extending-laika/03-implementing-directives.html#link-directives

**Comment by pme123** [comment]
Thanks, is there a simple example (not code snippets) on using the API Library. I struggle a bit in mixing SBT Plugin and the API Library. Resp. how to run it if I use only the API Library.

**Reply by jenshalm** [reply]
The link in previous response shows you how to implement a link directive. This section https://planet42.github.io/Laika/0.18/05-extending-laika/03-implementing-directives.html#directive-registration shows how to register any kind of directive, either with the sbt plugin or the library API. You only need those two pieces, it's fairly simple.

**Comment by pme123** [comment]
Ok - I found it - I should have checked the IO examples as well - this is what I need:).
